terraform-build-user/main.tf

```diff
@@ -9,6 +9,7 @@ module "iam_user" {
     aws.images-staging-ssm    = aws.images-staging-ssm
   }
 
-  ssm_parameters = ["/cyhy/dev/users", "/ssh/public_keys/*"]
-  user_name      = "build-ubuntu-packer"
+  # This module fails unless there is at least one SSM parameter
+  ssm_parameters = ["dummy_ssm_parameter"]
```

Comment:
This resurfaced to me this old PR I hadn't finished: https://github.com/cisagov/ami-build-iam-user-tf-module/pull/53. I think if that PR is merged we can removed this undesirable necessity.

Reply:
Good call - let's get [that sucker](https://github.com/cisagov/ami-build-iam-user-tf-module/pull/53) merged and then @bra1ncramp can remove this.

Reply:
@bra1ncramp https://github.com/cisagov/ami-build-iam-user-tf-module/pull/53 has been merged. Please remove this item and we can verify that everything works as expected after a `terraform apply`.

Reply:
Do we still need the `user_name - "build-ubuntu-server-packer"`?

Reply:
Correct, we still want that `user_name`.

Reply:
I'm going to resolve this convo on @mcdonnnj's behalf, since https://github.com/cisagov/ubuntu-server-packer/pull/8/commits/9620fa4f6ace316bb61692da693b29b53a0245f2 does indeed address his concern above.

Reply:
@dav3r I have unresolved this because I never got confirmation that this change was applied. I'm not sure who's driving the Terraform side of this so @dav3r, @jsf9k , and @bra1ncramp did someone apply this change and confirm it cleanly applied? Did a workflow run occur post-apply?

Reply:
@dav3r or @bra1ncramp - Did you gents reapply the `terraform-build-user` code (with an `init -upgrade`) after cisagov/ami-build-iam-user-tf-module#53 was merged?  And then was a new AMI built after that?  I defintely did not.

Reply:
Sorry, I forgot about that part - thanks for the reminder!  I just updated the build user (had to `destroy`, then `apply`, then update the GitHub secrets).

I just kicked off a [re-run of the most-recent prerelease](https://github.com/cisagov/ubuntu-server-packer/actions/runs/6040371872) so that we can confirm the new build user is working as expected.

Reply:
Hmmm...OK.  As long as you ran the `terraform init -upgrade`.

Reply:
I think I now realize the issue.  When I initially created the build user, this repo was still called `ubuntu-packer`.  Since then, we have renamed it to `ubuntu-server-packer`.  That change modified the key used by the `terraform-build-user` to track the user's state data in the backend.   When I re-initialized this terraform just now, I neglected to migrate it from the old state (keyed off of `ubuntu-packer`), so it had no knowledge of the existing user and related resources.  Since I didn't do that (and didn't realize what the issue was at the time), I went scorched earth and started fresh.

Reply:
A new AMI was [successfully created](https://github.com/cisagov/ubuntu-server-packer/actions/runs/6040371872/job/16586855143).

@bra1ncramp - I ran the `terraform-post-packer` if you want to verify that newest AMI.

Reply:
I ran `terraform init -upgrade`, then `terraform apply`
Successfully stood up an Ubuntu Server instance with the latest AMI

Reply:
To clarify, when I was referring to terraform commands above, it was related to the `terraform-build-user` code.  The Terraform that @bra1ncramp is referring to above is the code on the test terraformer instance to redeploy the Ubuntu server instance.  So bottom line, we should now be all clear to resolve this conversation @mcdonnnj.

Reply:
Excellent! Thanks all for getting this finalized.